test/integration/targets/aws_eks/playbooks/roles/aws_eks/tasks/main.yml

```diff
@@ -0,0 +1,193 @@
+      register: setup_security_groups
+
+    - name: create EKS cluster
+      aws_eks:
+        name: "{{ eks_cluster_name }}"
+        security_groups: "{{ eks_security_groups | json_query('[].name') }}"
+        subnets: "{{ setup_subnets.results | json_query('[].subnet.id') }}"
+        role_arn: "{{ iam_role.arn }}"
+        <<: *aws_connection_info
+      register: eks_create
+
+    - name: check that EKS cluster was created
+      assert:
+        that:
+          - eks_create is changed
+          - eks_create.name == eks_cluster_name
+
+    - name: create EKS cluster with same details but using SG ids
+      aws_eks:
+        name: "{{ eks_cluster_name }}"
+        security_groups: "{{ setup_security_groups.results | json_query('[].group_id') }}"
+        subnets: "{{ setup_subnets.results | json_query('[].subnet.id') }}"
+        role_arn: "{{ iam_role.arn }}"
+        <<: *aws_connection_info
+      register: eks_create
+
+    - name: check that EKS cluster did not change
+      assert:
+        that:
+          - eks_create is not changed
+          - eks_create.name == eks_cluster_name
+
+    - name: remove EKS cluster
+      aws_eks:
+        name: "{{ eks_cluster_name }}"
+        state: absent
+        <<: *aws_connection_info
+      register: eks_delete
+
+    - name: check that EKS cluster was created
+      assert:
+        that:
+          - eks_delete is changed
+
+  always:
+    - name: Announce teardown start
+      debug:
+        msg: "***** TESTING COMPLETE. COMMENCE TEARDOWN *****"
+
+    - name: remove EKS cluster
+      aws_eks:
+        name: "{{ eks_cluster_name }}"
+        state: absent
+        <<: *aws_connection_info
+      register: eks_delete
+      ignore_errors: yes
+
+    - debug:
+        msg: "{{ eks_security_groups|reverse|list }}"
+
+      # FIXME: bug in ec2_group means that co-dependent security groups appear to delete but don't!
+    - name: purge security group rules (co-dependency means they don't delete)
+      ec2_group:
+        name: '{{ item.name }}'
+        description: "{{ item.description }}"
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+        rules: []
+        rules_egress: []
+        purge_rules: yes
+        purge_rules_egress: yes
+        <<: *aws_connection_info
+      with_items: "{{ eks_security_groups|reverse|list }}"
+      ignore_errors: yes
+
+    - name: remove security groups
+      ec2_group:
+        name: '{{ item.name }}'
+        state: absent
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+        <<: *aws_connection_info
+      with_items: "{{ eks_security_groups|reverse|list }}"
+      ignore_errors: yes
+
+    - name: remove setup subnet
+      ec2_vpc_subnet:
+        az: '{{ aws_region }}{{ item.zone }}'
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+        cidr: "{{ item.cidr}}"
+        state: absent
+        <<: *aws_connection_info
+      with_items: "{{ eks_subnets }}"
+      ignore_errors: yes
+
+    - name: remove setup VPC
```

Comment:
Can you add some retries to this? I've run it a few times and gotten 
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_gfbyo1jp/ansible_module_ec2_vpc_net.py", line 467, in main
    connection.delete_vpc(VpcId=vpc_id)
  File "/tmp/ansible_gfbyo1jp/ansible_modlib.zip/ansible/module_utils/aws/core.py", line 241, in deciding_wrapper
    return unwrapped(*args, **kwargs)
  File "/Users/shertel/Workspace/ansible/venv/python3.6.0/lib/python3.6/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/shertel/Workspace/ansible/venv/python3.6.0/lib/python3.6/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (DependencyViolation) when calling the DeleteVpc operation: The vpc 'vpc-18020e61' has dependencies and cannot be deleted.
```
and had to wait and then delete the VPC myself.

Reply:
This is an ec2_group bug.